core/src/main/java/smile/regression/OLS.java

```diff
@@ -376,6 +391,54 @@ public double predict(double[] x) {
 
         return b + Math.dot(x, w);
     }
+    
+    /**
+     * Learn a new instance with online regression.
+     * @param x the training instance.
+     * @param y the target value.
+     */
+    @Override
+    public void learn(double[] x, double y){
+        JMatrix X = new JMatrix(p+1,1);
+        JMatrix W = new JMatrix(p+1,1);
+        for (int i=0; i<p; i++){
+            X.set(i,0,x[i]);
+            W.set(i,0,w[i]);
+        }
+        X.set(p,0,1);
+        W.set(p,0,b);
+        updateGamma(X);
+        updateW(X,W,y);
+    }
+    
+    private void updateGamma(JMatrix x){
+        double v = 1+gamma.xax(x.data());
```

Comment:
please add space between operator and parameters. Please keep consistent with the style of existing code. For example, we would like to format it as

```
private void updateGamma(JMatrix x) {
        double v = 1 + gamma.xax(x.data());
```